tools/cooja/apps/avrora/build.gradle

```diff
@@ -0,0 +1,24 @@
+apply plugin: "java"
+
+jar.baseName = 'cooja_avrora'
+
+repositories {
+  jcenter()
+}
+
+sourceSets {
+  main {
+    java {
+      srcDir 'src'
+    }
+  }
+
+}
+
+
+dependencies {
+    compile project(":")
+    compile files("lib/avrora-small.jar")
```

Comment:
> you can tell Gradle to always reload the jars from the repo even if their versions did not change (so your build always see the latest jar)

This will not fix the issue, as the problem is that the repo contains the wrong version.

> Their exact locations do not matter for the projects that depend on it, why should they "know" it?

You didn't get the point. I need to worry where Gradle gets that library, because I need that exact version, which however can not be distinguished from the wrong version by name and version, but _only_ by its location. Yet, if you don't believe me I might actually spend the time to break the approach once its implemented. I have spend enough time fighting this problem, because I had multiple checkouts of the same project and someone was of a the opinion that maven-artifacts are cool and created as many different artifacts as possible. It was a PITA to build two versions of the main application with a change somewhere down the dependency chain. I ended up giving them different version numbers which had to be propageted all the way up the dependency-chain.

Overall: Your solution is good for somewhat independent projects, like libraries providing functionality that might also be useful in another project. This is not the case here. What you should look into is https://docs.gradle.org/current/userguide/userguide_single.html#multi_project_builds . That approach described there might actually improve things and @sieben's current approach, but I'm not sure whether it is worth the trouble.